pyiode/objects/equation.pyx

```diff
@@ -217,6 +228,11 @@ cdef class Equation:
             raise TypeError("Expected value of type string.\nGot value of type '" + type(format).__name__ + "'")     
         return self.c_equation.get_date_as_string(format.encode()).decode()
 
+    # GDM> rename to get_coefficients? or even a .coefficients (readonly) property and move the "create missing scalars
```

Comment:
You raised a very good point here. 

@jmpplan please correct me if I am wrong.
The IODE function `@sliste` which is linked to the C function RPF_vsliste() from the source file api\report\engine\b_rep_fns.c. It returns the list of scalars (= coefficients) appearing in an equation but does not create entries in the global Scalars workspace for the missing scalars.

However, in the part dealing with estimations and in the code of the old GUI, I found some copy/paste code to 1) extract the list of scalars used in an equation and 2) create scalars with default value 0.9 and default relax 1.0 in the global Scalars workspace if not found.

So, I decided to create my own C++ method get_coefficients_list() with a boolean argument to whether or not create scalars entries if missing in the global workspace. 

I plan to one day rewrite the code of the Qt GUI in Python and to call the Python IODE API. 
So the best may be to, as you suggest, to create a `coefficients` property and to implement an hidden method You raised a very good point here. 

@jmpplan please correct me if I am wrong.
The IODE function `@sliste` which is linked to the C function RPF_vsliste() from the source file api\report\engine\b_rep_fns.c. It returns the list of scalars (= coefficients) appearing in an equation but does not create entries in the global Scalars workspace for the missing scalars.

However, in the part dealing with estimations and in the code of the old GUI, I found some copy/paste code to 1) extract the list of scalars used in an equation and 2) create scalars with default value 0.9 and default relax 1.0 in the global Scalars workspace if not found.

So, I decided to create my own C++ method get_coefficients_list() with a boolean argument to whether or not create scalars entries if missing in the global workspace. 

I plan to one day rewrite the code of the Qt GUI in Python and to call the Python IODE API. 
So the best may be to, as you suggest, to create a `coefficients` property and to implement an hidden method _get_and_create_coefficients()

Reply:
Implemented a *coefficients* property + a hidden method _get_and_create_coefficients() in PR #580